src/engine/sidechain/enginesidechain.cpp

```diff
@@ -75,11 +75,24 @@ void EngineSideChain::addSideChainWorker(SideChainWorker* pWorker) {
     m_workers.append(pWorker);
 }
 
-void EngineSideChain::writeSamples(const CSAMPLE* newBuffer, int buffer_size) {
+void EngineSideChain::receiveBuffer(AudioInput input,
```

Comment:
the name implies, that a pointer to the buffer is stored.
I have not yet understood why the additional check is rewired. Do you have name that explains this?